R/biome.R

```diff
@@ -0,0 +1,181 @@
+#'
+#' @param core Hector core
+#' @param old_biome Name of biome that will be split up
+#' @param new_biomes Names of biomes to be created
+#' @param fveg_c Fractions of vegetation C distributed to each biome
+#'   in `new_biomes`. Must be the same length as `new_biomes`. Default
+#'   is to split vegetation C evenly between all biomes.
+#' @param fdetritus_c Fractions of detritus C distributed to each
+#'   biome. Defaults to the same value as `fveg_c`.
+#' @param fsoil_c Fractions of soil C distributed to each biome.
+#'   Defaults to the same value as `fveg_c`.
+#' @param fnpp_flux0 Fraction of initial NPP flux distributed to each
+#'   biome. Defaults to the same value as `fveg_c`.
+#' @param ... Additional biome-specific parameters, as set by
+#'   [create_biome()]. Note that these are passed to [create_biome()]
+#'   via [base::mapply()], so they can be vectorized across biomes.
+#' @export
+split_biome <- function(core,
+                        old_biome,
+                        new_biomes,
+                        fveg_c = rep(1 / length(new_biomes), length(new_biomes)),
+                        fdetritus_c = fveg_c,
+                        fsoil_c = fveg_c,
+                        fnpp_flux0 = fveg_c,
+                        ...) {
+
+  stopifnot(
+    length(old_biome) == 1,
+    old_biome %in% get_biome_list(core),
+    !any(new_biomes %in% get_biome_list(core)),
+    length(fveg_c) == length(new_biomes),
+    length(fdetritus_c) == length(new_biomes),
+    length(fsoil_c) == length(new_biomes),
+    length(fnpp_flux0) == length(new_biomes),
+    sum(fveg_c) == 1, all(fveg_c > 0),
+    sum(fdetritus_c) == 1, all(fdetritus_c > 0),
+    sum(fsoil_c) == 1, all(fsoil_c > 0),
+    sum(fnpp_flux0) == 1, all(fnpp_flux0 > 0)
+  )
+
+  current_values <- get_biome_inits(core, old_biome)
+
+  # This allows users to split the `global` biome without having to
+  # rename it first. Otherwise, we hit an error about using non-global
+  # biomes when a "global" biome is present.
+  if (old_biome == "global") {
+    invisible(rename_biome(core, "global", "_zzz"))
+    old_biome <- "_zzz"
+  }
+
+  mapply(
+    create_biome,
+    biome = new_biomes,
+    veg_c0 = current_values[["veg_c"]] * fveg_c,
+    detritus_c0 = current_values[["detritus_c"]] * fdetritus_c,
+    soil_c0 = current_values[["soil_c"]] * fsoil_c,
+    npp_flux0 = current_values[["npp_flux0"]] * fnpp_flux0,
+    ...,
+    MoreArgs = list(core = core)
+  )
+
+  c_delete_biome(core, old_biome)
+
+  reset(core, 0)
+  invisible(core)
+
+}
+
+#' Merge multiple biomes into a single biome
+#'
+#' @param core Hector core
+#' @param old_biomes Names of biomes to be merged
+#' @param new_biome Name of new merged biome
+#' @param ... Additional parameters for the new biome, as set by
+#'   [create_biome()].
+#' @return `core`, invisibly
+#' @author Alexey Shiklomanov
+#' @export
+merge_biomes <- function(core, old_biomes, new_biome, ...) {
```

Comment:
Maybe I'm not imaginative enough...what's the use case for this? I.e. when would the user ever set up biomes, and then merge then?

Reply:
In the long term, I was thinking it might be possible to alter the number of biomes dynamically (e.g. have biomes that only exist during a certain point in time). But that's not possible the way this is currently implemented, and might not make sense as an idea anyway, so I think you're right that this is unnecessary. I'll remove it.

Reply:
Removed in 70bca145c61ce0a0d91764c1383140fbecb94006.